### PR TITLE
More server enum lists (plus PotionEffectType), fixes #1916

### LINKED ISSUE #1916: More enum lists
Tree types: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/TreeType.html
Map cursor types: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/map/MapCursor.Type.html
Potion types: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/potion/PotionType.html
Potion effect types: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/potion/PotionEffectType.html
Entity types: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/EntityType.html
Banner/shield patterns: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/block/banner/PatternType.html

I had also found structure types, but since Denizen doesn't really have much things related to structures, I dunno if I should add them yet. Will link it just in case: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/StructureType.html

### PR DESCRIPTION
**Pull request by MusicScore**
Tested to work on all supportable versions (1.8.8 to 1.13.2).

### CODE REVIEW

**Comment by mcmonkey4eva** [comment]
Would be good to link to the new tags for places they're relevant (cast command meta reference the potion tags, spawn command reference entity type list, etc)

There's also a potential question of whether entity_types should include Denizen custom types like `fake_player` and `item_projectile`, but I think it's better to not include them in this case (as they're only useful for certain Denizen functionalities, and also do not vary between server versions, rather by Denizen version).

Also please add `fixes #1916` to the PR title or any commit title (that's GitHub's way of linking PRs/commits/issues together).

**Comment by MusicScore** [comment]
Did what you asked, and threw in two more enum lists because why not.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 1]
```diff
@@ -877,10 +878,7 @@ public void registerCoreMembers() {
         // This command creates a new minecraft world with the specified name, or loads an existing world by thet name.
         // TODO: Document Command Details (generator)
         // It accepts a world type which can be specified with 'worldtype:'.
-        // If a worldtype is not specified it will create a world with a worldtype of NORMAL.
-        // Recognised world type are NORMAL (creates a normal world), FLAT (creates a world with flat terrain),
-        // LARGE_BIOMES (creates a normal world with 16x larger biomes) and AMPLIFIED (creates a world with tall
-        // mountain-like terrain).
+        // If a worldtype is not specified it will create a world with a world type of NORMAL.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java:881]: Did you forget to add in a link here?

Comment (ghost) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java:881]: Oops

**Pull request merged**